### DISCUSSION TITLE
How to run only the IB Gateway

### DISCUSSION THREAD

**Discussion by edgebroker** [gnzsnz/ib-gateway-docker#165] [Q&A] [answered]
Hi,

first of all, thanks for this!

I only need the IB gateway but not TWS and VNC. Is there a way to disable everything except IB gateway?

Thanks,
Andreas

**Comment by gnzsnz** [comment, marked as answer]
To run ibgateway or TWS you just need to chose the right image, you can't run both (not that i'm aware).

Regarding VNC, it only works for ibgateway (TWS image comes with rdesktop) there is a setting to enable/disable it, it's `VNC_SERVER_PASSWORD` if you set a password, then it will start, if you don't set a password then it wont start. it's on the [README](https://github.com/gnzsnz/ib-gateway-docker) with all the other settings.

**Reply by JohnAllen** [reply]
So there's no way to even see TWS instead of Gateway @gnzsnz ? Why is this out of curiosity (if so)? Would be helpful to be able to just see TWS and I would consider working on it if it wasn't too demanding and I can learn whatever it is I need to do.

**Reply by gnzsnz** [reply]
not sure what you mean, you can run TWS instead of ibgateway. you just need to use the sample tws docker compose as a starting point, there is a section on the README with TWS specifics

**Reply by JohnAllen** [reply]
Sorry @gnzsnz I meant about the VNC server. That only works with Gateway? That's what I was actually asking about.

**Reply by gnzsnz** [reply]
> Sorry @gnzsnz I meant about the VNC server. That only works with Gateway? That's what I was actually asking about.

TWS runs using remote desktop
